### PR TITLE
chore: rewrite deploy action to be able to deploy multiple versions t…

### PR DESCRIPTION
**Pull request by joyenjoyer**
…o gh pages

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -1,19 +1,40 @@
...
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
       - run: npm ci && npm run bootstrap
+      # Conditional build depending on branch
+      - name: Set build directory based on branch
+        id: set-build-dir
+        run: |
+          if [ "${{ github.ref }}" == "refs/heads/master" ]; then
+            echo "::set-output name=build_dir::./packages/__docs__/__build__"
```
Comment (balzss) [inline .github/workflows/deploy.yml:20]: could this be set before the if/elif statement? it's the same for every condition

File: packages/__docs__/_redirects_gh_pages
Comment (balzss) [inline packages/__docs__/_redirects_gh_pages:1]: is this file needed? isn't it for netflity redirects?

File: packages/__docs__/src/Header/index.tsx [hunk 2]
```diff
@@ -69,9 +69,12 @@ class Header extends Component<HeaderProps> {
     // If we select the latest version from the dropdown,
     // then navigate to the index (instructure.design/#currentHash).
     // In every other case eg.: v6,v7 navigate to --> instructure.design/v6/#currentHash
+    const rootToAdd = window.location.origin.includes('github.io')
```
Comment (balzss) [inline packages/__docs__/src/Header/index.tsx:72]: please add and explanatory comment for this. could be a TODO with something like "remove when we permanently migrate to github.io"

File: packages/__docs__/src/versionData.js [hunk 3]
```diff
@@ -33,17 +33,14 @@
...
 
   if (!isLocalHost) {
-    // eslint-disable-next-line compat/compat
-    const result = await fetch(`${window.location.origin}/versions.json`, {
-      signal
-    })
-    const versionsData = await result.json()
-
-    return versionsData
+    let input = window.location.hostname.includes('github.io')
```
Comment (balzss) [inline packages/__docs__/src/versionData.js:39]: please add an explanation here too, maybe with a TODO remove later

File: packages/__docs__/src/versionData.js [hunk 4]
```diff
@@ -53,7 +50,9 @@ const fetchVersionData = async (signal) => {
  * if we are on the docs page of a legacy version,
  * the path includes the version number, e.g. `/v7` or `/v8`
  */
-const [versionInPath] = window.location.pathname.split('/').filter(Boolean)
+const lastSegment =
+  window.location.pathname.split('/').filter(Boolean).pop() || ''
+const versionInPath = /v\d$/.test(lastSegment) ? lastSegment : undefined
```
Comment (balzss) [inline packages/__docs__/src/versionData.js:55]: these 2 lines are not really straight forward. what do they do?

**Review by balzss — CHANGES_REQUESTED** [review]
please add some explanations to the code, otherwise nice work!

**Review by HerrTopi — APPROVED** [review]

**Review by matyasf — APPROVED** [review]

**Review by balzss — APPROVED** [review]

**Pull request merged**